File: 2024/aoc01.py
import math
import adj
ll = open("input.txt",'r').read().splitlines()
l1 = []
l2 = []
for l in ll:
    l = l.split('   ')
    l1 += [int(l[0])]
    l2 += [int(l[1])]

l1 = list(sorted(l1))
l2 = list(sorted(l2))

a  = 0
for i in range(len(l1)):
    a += abs(l1[i]-l2[i])


print(a)

a  = 0
for i in l1:
    a += i*l2.count(i)

print(a)

File: 2024/aoc01.rs
use std::fs;

fn main() {
    let contents = fs::read_to_string("../input.txt").expect("");
    let split = contents.split("\n");
    let mut l1 = Vec::new();
    let mut l2 = Vec::new();
    let mut a = 0;
    for line in split {
        if line != "" {
            let num: Vec<i32> = line.split("   ").map(|x| x.parse::<i32>().unwrap()).collect();
            l1.push(num[0]);
            l2.push(num[1]);
        }
    }
    l1.sort();
    l2.sort();
    for i in 0..l1.len() {
        a += (l1[i]-l2[i]).abs();
    }
    println!("{a}");


    a = 0;
    for i in 0..l2.len() {
        if l1.contains(&l2[i]){
            a += l2[i]
        }
    }
    println!("{a}")
}
